src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -404,7 +404,18 @@ private void BuildProjectWithMsbuild()
             {
                 if (property.Value.Contains(" "))
                 {
-                    properties += $" /p:{property.Key}=\"{property.Value}\"";
+                    var length = property.Value.Length;
+                    // If a property value ends with a '\' and has a space, then the closing quote gets escaped because of the '\' . In order to overcome this,
+                    // we add an extra '\' for safety. This issue came up with bug : https://github.com/NuGet/Home/issues/3432
+                    if (property.Value[length - 1] == '\\' && length >=2 && property.Value[length - 2] != '\\')
```

Comment:
`.Value[length - 1]` can be `EndsWith`?
`length >=2` needs an extra space before the 2

This seems like a generic escaping problem. Perhaps we should extract this logic to another method and test more scenarios. For example, what happens when the `property.Value` has quotes in it? 

Coincidentally, I literally worked on this for one of my own projects over the weekend:
https://github.com/joelverhagen/Procommand/blob/34e78d1d941bcc913c6fd45fd42ffea8394a53ee/src/Knapcode.Procommand/ArgumentsBuilder.cs#L57-L91

Inspired by StackOverflow answer: http://stackoverflow.com/a/12364234